patches/api/0461-add-rich-message-component-support-to-configuration.patch

```diff
@@ -0,0 +1,68 @@
+
+
+diff --git a/src/main/java/org/bukkit/configuration/ConfigurationSection.java b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+index b6b00af08f12f838411845e4f4e29e62826dfc7f..911eff863241373462eaa3e420ae315a8df4980b 100644
+--- a/src/main/java/org/bukkit/configuration/ConfigurationSection.java
++++ b/src/main/java/org/bukkit/configuration/ConfigurationSection.java
+@@ -1058,4 +1058,57 @@ public interface ConfigurationSection {
+      * one line.
+      */
+     public void setInlineComments(@NotNull String path, @Nullable List<String> comments);
++
++    // Paper start - add rich message component support to configuration
++    /**
++     * Gets the requested MiniMessage formatted String as Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @return Requested Component.
++     */
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path) {
++        return getRichMessage(path, null);
++    }
++
++    /**
++     * Gets the requested MiniMessage formatted String as Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param fallback component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, !null -> !null")
++    @Nullable
++    default net.kyori.adventure.text.Component getRichMessage(@NotNull String path, @Nullable net.kyori.adventure.text.Component fallback) {
++        return getComponent(path, net.kyori.adventure.text.minimessage.MiniMessage.miniMessage(), fallback);
++    }
++
++    /**
++     * Gets the requested MiniMessage formatted String as Component by path.
++     * <p>
++     * If the Component does not exist but a default value has been specified,
++     * this will return the default value. If the Component does not exist and no
++     * default value was specified, this will return null.
++     *
++     * @param path Path of the Component to get.
++     * @param componentSerializer componentSerializer instance used for deserialization
++     * @param fallback component that will be used as fallback
++     * @return Requested Component.
++     */
++    @Contract("_, _, !null -> !null")
++    @Nullable
++    default <C extends net.kyori.adventure.text.Component> C getComponent(@NotNull String path, @NotNull net.kyori.adventure.text.serializer.ComponentSerializer<C, C, String> componentSerializer, @Nullable C fallback) {
```

Comment:
I think the first generic param should be `?` cause we technically don't care at all about that input type, we only care about the output type.

Reply:
> I think the first generic param should be `?` cause we technically don't care at all about that input type, we only care about the output type.

Although if we wait for ComponentDecoder to be a thing this wouldn't be a concern

Reply:
Addressed by using a local version of https://github.com/KyoriPowered/adventure/pull/1035 see https://github.com/PaperMC/Paper/pull/10225/commits/45b2e17f141e167b861f15b21dc9c85277fb5786